src/rgw/rgw_rest_s3.cc

```diff
@@ -292,6 +292,11 @@ int RGWGetObj_ObjStore_S3::get_params(optional_yield y)
   // all of the data from its parts. the parts will sync as separate objects
   skip_manifest = s->info.args.exists(RGW_SYS_PARAM_PREFIX "sync-manifest");
 
+  string checksum_mode_arg = s->info.env->get("HTTP_CHECKSUM_MODE","");
```

Comment:
this is from the [x-amz-checksum-mode](https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetObject.html#API_GetObject_RequestSyntax) header, so "HTTP_X_AMZ_CHECKSUM_MODE"